4-citation/check_citation.py

```diff
@@ -1,16 +1,54 @@
-def first_check():
-    pass
+import os.path
+from os import path
+import yaml
+from voluptuous import Schema, MultipleInvalid, Invalid, Required, Any, Optional, ALLOW_EXTRA
 
+def file_check(filename):
+	if path.isfile(filename):
+		print ("File " +  filename + " found")
+		return "File Found"
+	else: 
+		print("File " + filename + " not found")
+		return "File Not Found"
 
-def second_check():
-    pass
 
 
-def main():
-    print('citation checks TODO')
-    first_check()
-    second_check()
+def YAML_check(filename):
+    with open(filename) as file:
+	try:
+		citation = yaml.safe_load(file)
+		return "YAML file loaded"
+	except yaml.YAMLError as exc:
+        	print(exc)
+		return "YAML failed to load"
+
+define Contents_check(filename):
+	citation = yaml.safe_load(file)
+	s = Schema({
+		Required("doi"): str,
+		Required("title"): str,
+		Optional("version"): str,
+		Required("authors"): str,
```

Comment:
There will be an issue here, because `authors` is not a string but a dictionary.